units/commit_queue.go

```diff
@@ -104,165 +104,175 @@ func (j *commitQueueJob) Run(ctx context.Context) {
-			"project": j.QueueID,
-			"item":    nextItem,
-			"message": "can't dequeue item",
-		}))
+	modulePRs, modulePatches, dequeue, err := getModules(ctx, githubToken, nextItem, projectConfig)
+	if err != nil {
+		j.logError(err, "can't get modules", nextItem)
+		if dequeue {
+			j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't get modules", ""))
+			j.dequeue(cq, nextItem)
+		} else {
+			j.AddError(errors.Wrap(cq.SetProcessing(false), "can't set processing to false"))
+		}
 		return
 	}
+	patchDoc.Patches = modulePatches
 
-	// GitHub hasn't yet tested if the PR is mergeable.
-	// Check back later
-	// See: https://developer.github.com/v3/pulls/#response-1
-	if pr.Mergeable == nil {
+	v, err := makeVersion(ctx, githubToken, projectConfig, patchDoc)
+	if err != nil {
+		j.logError(err, "can't make version", nextItem)
+		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make version", ""))
+		j.dequeue(cq, nextItem)
 		return
 	}
 
-	if !*pr.Mergeable {
-		err = sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "PR not mergeable", "")
+	j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStatePending, "preparing to test merge", v.Id))
+	for _, modulePR := range modulePRs {
+		j.AddError(sendCommitQueueGithubStatus(j.env, modulePR, message.GithubStatePending, "preparing to test merge", v.Id))
+	}
+
+	err = subscribeMerge(projectRef.Identifier, projectRef.Owner, projectRef.Repo, projectRef.CommitQueue.MergeMethod, v.Id, pr)
+	if err != nil {
+		j.logError(err, "can't subscribe GitHub PR merge to version", nextItem)
+		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't start merge", ""))
+		j.dequeue(cq, nextItem)
+	}
+
+	for _, modulePR := range modulePRs {
+		err = subscribeMerge("", modulePR.Base.Repo.Owner.GetLogin(), *modulePR.Base.Repo.Name, projectRef.CommitQueue.MergeMethod, v.Id, modulePR)
 		if err != nil {
-			j.AddError(errors.Wrap(err, "can't send github status"))
+			j.logError(err, "can't subscribe GitHub PR merge to version for module", nextItem)
+			j.AddError(sendCommitQueueGithubStatus(j.env, modulePR, message.GithubStateFailure, "can't start merge", ""))
 		}
-		return
 	}
 
-	// check if a patch has already been created for this PR
-	mergeCommitSHA := pr.GetMergeCommitSHA()
-	existingPatch, err := patch.FindOneByGithubPRNumAndMergeCommitSHA(nextItemInt, mergeCommitSHA)
-	if err != nil {
-		j.AddError(errors.Wrap(err, "can't query for patch matching PR Number"))
+	grip.Info(message.Fields{
+		"source":  "commit queue",
+		"job_id":  j.ID(),
+		"item":    nextItem,
+		"message": "finished processing item",
+	})
+}
+
+func (j *commitQueueJob) logError(err error, msg string, item *commitqueue.CommitQueueItem) {
+	if err == nil {
 		return
 	}
-	if existingPatch != nil {
-		return
+	j.AddError(errors.Wrap(err, msg))
+	grip.Error(message.WrapError(err, message.Fields{
+		"job":     commitQueueJobName,
```

Comment:
maybe good to put j.ID() in this message...

Reply:
done.